portal-kernel/src/com/liferay/portal/kernel/workflow/WorkflowEngineManagerUtil.java

```diff
@@ -25,28 +24,32 @@
 public class WorkflowEngineManagerUtil {
 
 	public static String getKey() {
-		return _workflowEngineManager.getKey();
+		return _KEY;
 	}
 
 	public static String getName() {
-		return _workflowEngineManager.getName();
+		return _NAME;
 	}
 
 	public static Map<String, Object> getOptionalAttributes() {
-		return _workflowEngineManager.getOptionalAttributes();
+		return _optionalAttributes;
 	}
 
 	public static String getVersion() {
-		return _workflowEngineManager.getVersion();
+		return _VERSION;
 	}
 
 	public static boolean isDeployed() {
-		return _workflowEngineManager.isDeployed();
+		return true;
 	}
 
-	private static volatile WorkflowEngineManager _workflowEngineManager =
-		ServiceProxyFactory.newServiceTrackedInstance(
-			WorkflowEngineManager.class, WorkflowEngineManagerUtil.class,
-			"_workflowEngineManager", true);
+	private static final String _KEY = "liferay";
+
+	private static final String _NAME = "Liferay Kaleo Workflow Engine";
```

Comment:
@JanisZhang I don't think it's right to inline this. This engine name is "Kaleo", which integrations the third-party Kaleo library into Liferay. So if there's another third-party lib called "ABC" we can also have a `WorkflowEngineManagerImpl` that returns "Liferay ABC Workflow Engine" as name.

I think the best way is to keep the interface and the implementation, but remove `isDeployed()` method from them. Instead of returning the boolean directly from the impl, do a check to return true/false based on the existence of the `WorkflowEngineManager` service.

And remove the other methods from this util, make it an `registry` of `WorkflowEngineManager`s.